modules/apps/export-import/export-import-service/src/main/java/com/liferay/exportimport/internal/controller/PortletExportControllerImpl.java

```diff
@@ -1224,14 +1223,24 @@ private String _getLockPath(
 	private PortletDataHandler _getPortletDataHandler(
 		PortletDataContext portletDataContext, Portlet portlet) {
 
-		Optional<Portlet> portletOptional = _replacePortlet(
-			portletDataContext, portlet);
+		if (ExportImportDateUtil.isRangeFromLastPublishDate(
+				portletDataContext)) {
+
+			String changesetPortletId = ChangesetPortletKeys.CHANGESET;
+
+			if (ExportImportThreadLocal.isPortletStagingInProcess()) {
```

Comment:
@JanisZhang To make the logic consistent with before, you need to do 
f (ExportImportThreadLocal.isPortletStagingInProcess()) {
				portlet = _portletLocalService.getPortletById(
					changesetPortletId);
			}
**else** if (ExportImportThreadLocal.isLayoutStagingInProcess() &&
				!changesetPortletId.equals(portlet.getPortletId())) {

				return null;
			}

Reply:
@JanisZhang This is hard to figure out as you modify and merge in the logic in one commit, separating into two commits may help to know trap like this.